implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/MetricsSender.java

```diff
@@ -24,9 +24,12 @@ interface MetricsSender {
     /**
      * Send encoded metrics data from a {@link io.micrometer.core.instrument.MeterRegistry
      * MeterRegistry}.
+     * @param address where to send the metrics
      * @param metricsData encoded batch of metrics
      * @param headers metadata to send as headers with the metrics data
+     * @throws Throwable when there is an exception in sending the metrics; the caller
+     * should handle this in some way such as logging the exception
      */
-    void send(byte[] metricsData, Map<String, String> headers);
+    void send(String address, byte[] metricsData, Map<String, String> headers) throws Throwable;
```

Comment:
I didn't mention it in the PR description but `throws Throwable` was added here and elsewhere. The point was to ensure that the caller (MeterRegistry publish implementation) handles the exception case, usually with logging. `HttpSender` similarly declares `throws Throwable`. That said, it may be worth looking at doing things differently here because catching Throwable includes things like OOME, as pointed out by @marcingrzejszczak in an offline review.